### PR TITLE
uar-133 resigned on & ch references

### PR DESCRIPTION
**Pull request by AndreKramer7654**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-133

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerCorporateValidator.java [hunk 1]
```diff
@@ -1,11 +1,14 @@
 package uk.gov.companieshouse.overseasentitiesapi.validation;
 
+import java.time.LocalDate;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
 import uk.gov.companieshouse.overseasentitiesapi.model.dto.ManagingOfficerCorporateDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.ManagingOfficerIndividualDto;
```
Comment (corybudd-ch) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerCorporateValidator.java:8]: From sonarqube: "remove this unused import 'uk.gov.companieshouse.overseasentitiesapi.model.dto.ManagingOfficerIndividualDto'."

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java [hunk 2]
```diff
@@ -44,6 +44,9 @@ public class BeneficialOwnerGovernmentOrPublicAuthorityDao {
     @Field("ceased_date")
     private LocalDate ceasedDate;
 
+    @Field("ch_reference")
```
Comment (stephanie-ch) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java:47]: ch_reference looks incorrect here?

Comment (stephanie-ch) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java:47]: Okay, looks the same in the other files. So ch_reference is chips reference - but is that also the ID in self.links?

**Review by jkayyala — APPROVED** [review]
Other than the unused import , looks fine

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java (hunk 2 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dao/BeneficialOwnerGovernmentOrPublicAuthorityDao.java:47]: It's part (last component) of the ID in self.links

**Review by stephanie-ch — APPROVED** [review]

**Review by corybudd-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerCorporateValidator.java [hunk 3]
```diff
@@ -52,6 +54,15 @@ public Errors validate(List<ManagingOfficerCorporateDto> managingOfficerCorporat
             validateRoleAndResponsibilities(managingOfficerCorporateDto.getRoleAndResponsibilities(), errors, loggingContext);
             validateContactFullName(managingOfficerCorporateDto.getContactFullName(), errors, loggingContext);
             validateContactEmail(managingOfficerCorporateDto.getContactEmail(), errors, loggingContext);
+
+            if (managingOfficerCorporateDto.getStartDate() != null) {
+                validateStartDate(managingOfficerCorporateDto.getStartDate(), errors, loggingContext);
+                validateResignedOnDateAgainstStartDate(managingOfficerCorporateDto.getResignedOn(), managingOfficerCorporateDto.getStartDate(), errors, loggingContext);
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerCorporateValidator.java:60]: there is no null check performed on getResignedOn() so the resigned_on date could potentially be null on line 60. the validateResignedOnDateAgainstStartDate method will pass the resigned_on date into DateValidators.isDateInPast, which will try and call .isAfter() on the null resigned_on (if null at this point) causing a null pointer exception. You'd need an extra null check against resigned_on before comparing against the start_date to make sure both dates are not null.

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/ManagingOfficerCorporateValidator.java:60]: Good spot and my bad (as I had the null check in the else).

**Review by markpit — APPROVED** [review]

**Pull request merged**